src/swtpm/utils.c

```diff
@@ -329,3 +334,43 @@ ssize_t read_eintr(int fd, void *buffer, size_t buflen)
         return n;
     }
 }
+
+/*
+ * disable_fips_mode: If possible, disable FIPS mode to avoid libtpms failures
+ *
+ * While libtpms does not provide a solution to disable deactivated algorithms
+ * avoid libtpms failures due to FIPS mode enablement by disabling FIPS mode.
+ */
+#ifdef HAVE_OPENSSL_FIPS_H
+int disable_fips_mode(void)
+{
+    int mode = FIPS_mode();
+    int ret = 0;
+
+    if (mode != 0) {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+        logprintf(STDERR_FILENO,
+                  "FIPS mode cannot be disabled with OpenSSL 3.x\n");
```

Comment:
The openssl 3 'migration_guide' man page says

```
       Removed FIPS_mode() and FIPS_mode_set()

       These functions are legacy APIs that are not applicable to the new
       provider model. Applications should instead use
       EVP_default_properties_is_fips_enabled(3) and
       EVP_default_properties_enable_fips(3).
```
so we don't need to stub this out as an error AFAICT

Reply:
I ended up using the recommented EVP APIs...